Windows hidden file test failure

https://github.com/ianlewis/todos/actions/runs/5167637338/jobs/9308650495

```
--- FAIL: Test_isHidden/some\path\.foo.bar#01 (0.00s)
panic: open C:\Users\RUNNER~1\AppData\Local\Temp\utils_windows_test170354236\some\path\.foo.bar: The system cannot find the path specified. [recovered]
	panic: open C:\Users\RUNNER~1\AppData\Local\Temp\utils_windows_test170354236\some\path\.foo.bar: The system cannot find the path specified.

goroutine 432 [running]:
testing.tRunner.func1.2({0x1239660, 0xc00123a630})
	C:/hostedtoolcache/windows/go/1.20.4/x64/src/testing/testing.go:1526 +0x24e
testing.tRunner.func1()
	C:/hostedtoolcache/windows/go/1.20.4/x64/src/testing/testing.go:1529 +0x39f
panic({0x1239660, 0xc00123a630})
	C:/hostedtoolcache/windows/go/1.20.4/x64/src/runtime/panic.go:884 +0x213
github.com/ianlewis/todos/internal/testutils.Check(...)
	D:/a/todos/todos/internal/testutils/utils.go:20
github.com/ianlewis/todos/internal/cmd/todos/walker.Test_isHidden.func1(0xc0012a1040)
	D:/a/todos/todos/internal/cmd/todos/walker/utils_windows_test.go:[100](https://github.com/ianlewis/todos/actions/runs/5167500382/jobs/9308416258?pr=49#step:4:101) +0x29e
testing.tRunner(0xc0012a[104](https://github.com/ianlewis/todos/actions/runs/5167500382/jobs/9308416258?pr=49#step:4:105)0, 0xc00142b760)
	C:/hostedtoolcache/windows/go/1.20.4/x64/src/testing/testing.go:1576 +0x10b
created by testing.(*T).Run
	C:/hostedtoolcache/windows/go/1.20.4/x64/src/testing/testing.go:1629 +0x3ea
```

--- a/internal/cmd/todos/walker/utils_windows_test.go
+++ b/internal/cmd/todos/walker/utils_windows_test.go
@@ -97,7 +97,7 @@ func Test_isHidden(t *testing.T) {
 
 			dir := testutils.Must(os.MkdirTemp("", "utils_windows_test"))
 			path := filepath.Join(dir, tc.name)
-			testutils.Check(os.WriteFile(path, nil, 0600))
+			testutils.Check(os.WriteFile(path, nil, 0o666))
 			if tc.hiddenAttr {
 				testutils.Check(setHidden(path))
 			}

--- a/internal/cmd/todos/walker/utils_windows_test.go
+++ b/internal/cmd/todos/walker/utils_windows_test.go
@@ -97,7 +97,8 @@ func Test_isHidden(t *testing.T) {
 
 			dir := testutils.Must(os.MkdirTemp("", "utils_windows_test"))
 			path := filepath.Join(dir, tc.name)
-			testutils.Check(os.WriteFile(path, nil, 0o666))
+			testutils.Check(os.MkdirAll(filepath.Dir(path), 0o600))
+			testutils.Check(os.WriteFile(path, nil, 0o600))
 			if tc.hiddenAttr {
 				testutils.Check(setHidden(path))
 			}

